docs/guides/context.md

```diff
@@ -6,20 +6,24 @@ While _finite_ states are well-defined in finite state machines and statecharts,
 
 In XState, extended state is known as **context**. Below is an example of how `context` is used to simulate filling a glass of water:
 
-```js
-import { createMachine, assign } from 'xstate';
+```ts
+import { createMachine, assign, ConditionPredicate } from 'xstate';
+
+interface Context {
+  amount: number;
+}
+type Event = { type: 'FILL' };
 
 // Action to increment the context amount
-const addWater = assign({
-  amount: (context, event) => context.amount + 1
+const addWater = assign<Context, Event>({
+  amount: (context) => context.amount + 1
 });
 
 // Guard to check if the glass is full
-function glassIsFull(context, event) {
-  return context.amount >= 10;
-}
+const glassIsFull: ConditionPredicate<Context, Event> = (context) =>
```

Comment:
```suggestion
const glassIsFull = (context: Context) =>
```

Reply:
I thought it would be nice to show off the `ConditionPredicate` type, but can do the simple version too.

Reply:
> I thought it would be nice to show off the `ConditionPredicate` type, but can do the simple version too.

Let's do the simple version, as I think `ConditionPredicate` seems like extra things you need to know (mental overload) and the name of that type might change in v5 anyway (to `GuardPredicate` or something like that).